test/metabase/query_processor_test/advanced_math_test.clj

```diff
@@ -63,28 +65,46 @@
                    int)))))
 
 
-(defn- test-aggregation
-  [agg]
-  (->> {:aggregation [agg]}
-       (mt/run-mbql-query venues)
-       mt/rows
-       ffirst
-       double
-       (u/round-to-decimals 2)))
+(defn- aggregation=
+  [expected agg]
+  (testing "As a top-level aggregation"
+    (let [query (mt/mbql-query venues
+                  {:aggregation [agg]})]
+      (mt/with-native-query-testing-context query
+        (is (= expected
+               (ffirst
+                (mt/formatted-rows [1.0]
+                  (mt/process-query query))))))))
+  (when (driver/database-supports? driver/*driver* :expression-aggregations (mt/db))
+    (testing "Inside an expression aggregation"
+      (let [query (mt/mbql-query venues
+                    {:aggregation [[:+ agg 1]]})]
+        (mt/with-native-query-testing-context query
+          (is (= (+ expected 1.0)
+                 (ffirst
+                  (mt/formatted-rows [1.0]
+                    (mt/process-query query))))))))))
```

Comment:
I forgot that MongoDB did not support `:expression-aggregations` so I wanted to make sure that `:var` worked inside an expression aggregation (since it looked to me like the `expand-aggregation` stuff would not apply there). So I reworked this test util function to test both a top-level aggregation, and the aggregation inside an expression for drivers that support it. It also adds `testing` context with the compiled native query, which helped me debug MongoDB failures